system-parachains/people/people-kusama/src/identity_ops.rs

```diff
@@ -0,0 +1,182 @@
+				} else {
+					true
+				}
+			});
+
+			((judgements_count - identity.0.judgements.len()) as u32, identity)
+		}
+
+		/// Weight calculation for the worst-case scenario of `clear_judgement`.
+		/// Equal to 1 identity read/write + 1 subs read + 1 reserve balance read.
+		fn weight_clear_judgement() -> Weight {
+			<Runtime as frame_system::Config>::DbWeight::get().reads_writes(3, 1)
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		#[cfg(feature = "try-runtime")]
+		fn try_state(_: BlockNumberFor<T>) -> Result<(), sp_runtime::TryRuntimeError> {
+			use crate::Balance;
+			use sp_core::crypto::Ss58Codec;
+			use sp_runtime::traits::Zero;
+
+			let mut invalid_identity_count = 0;
+			let mut invalid_judgement_count = 0;
+			let mut identity_count = 0;
+			IdentityOf::iter().for_each(|(account_id, _)| {
+				let (identity, username) = IdentityPallet::identity(&account_id).unwrap();
+
+				let (paid_judgement_count, judgement_deposit) = identity.judgements.iter().fold(
+					(0, Zero::zero()),
+					|(count, total_deposit): (u32, Balance), (_, judgement)| {
+						if let Judgement::FeePaid(deposit) = judgement {
+							(count + 1, total_deposit.saturating_add(*deposit))
+						} else {
+							(count, total_deposit)
+						}
+					},
+				);
+
+				let (subs_deposit, _) = IdentityPallet::subs_of(&account_id);
+
+				let deposit =
+					identity.deposit.saturating_add(judgement_deposit).saturating_add(subs_deposit);
+				let deposit_wo_judgement = identity.deposit.saturating_add(subs_deposit);
+				let reserved = Balances::reserved_balance(&account_id);
+
+				if deposit > reserved && paid_judgement_count > 0 {
+					invalid_identity_count += 1;
+					invalid_judgement_count += paid_judgement_count;
+
+					log::info!(
+						"account with invalid state: {:?}, expected reserve at least: {:?}, actual: {:?}",
+						account_id.clone().to_ss58check(),
+						deposit,
+						reserved,
+					);
+
+					assert_eq!(
+						paid_judgement_count,
+						Self::do_clear_judgement(&account_id, (identity, username)).0
+					);
+
+					if deposit_wo_judgement != reserved {
+						log::warn!(
+							"unexpected state: {:?}, deposit w/o judgement: {:?}, not equal to the total reserved: {:?}",
+							account_id.clone().to_ss58check(),
+							deposit_wo_judgement,
+							reserved,
+						);
+					}
+				} else {
+					assert_eq!(0, Self::do_clear_judgement(&account_id, (identity, username)).0);
+				}
+				if deposit_wo_judgement > reserved {
+					log::warn!(
+						"unexpected state: {:?}, deposit w/o judgement: {:?}, greater than the total reserved: {:?}",
+						account_id.clone().to_ss58check(),
+						deposit_wo_judgement,
+						reserved,
+					);
+				}
+				identity_count += 1;
+			});
+
+			log::info!("total identities processed: {:?}", identity_count);
+			log::info!("invalid identities: {:?}", invalid_identity_count);
+			log::info!("invalid judgements: {:?}", invalid_judgement_count);
```

Comment:
when run for Kusama People Chain, this result the iterations over `4298` identities, clean up for `82` identities and `86` judgements.